xarray/test/test_dataarray.py

```diff
@@ -1210,6 +1213,106 @@ def test_groupby_first_and_last(self):
+
+        self.assertEqual(rolling_obj._axis_num, 0)
+
+        # catching invalid args
+        with self.assertRaisesRegexp(ValueError, 'only one dim/window can be'):
+            da.rolling(time=7, x=2)
+        with self.assertRaisesRegexp(ValueError, 'window must be > 0'):
+            da.rolling(time=-2)
+
+    @requires_bottleneck
+    def test_rolling_wrapped_bottleneck(self):
+        import bottleneck as bn
+
+        da = self.make_rolling_example_array()
+
+        # Test all bottleneck functions
+        rolling_obj = da.rolling(time=7)
+        for name in ('sum', 'mean', 'std', 'min', 'max', 'median'):
+            func_name = 'move_{0}'.format(name)
+            actual = getattr(rolling_obj, name)()
+            expected = getattr(bn, func_name)(da.values, window=7, axis=0)
+            self.assertArrayEqual(actual.values, expected)
+
+        # Using min_periods
+        rolling_obj = da.rolling(time=7, min_periods=1)
+        for name in ('sum', 'mean', 'std', 'min', 'max'):
+            func_name = 'move_{0}'.format(name)
+            actual = getattr(rolling_obj, name)()
+            expected = getattr(bn, func_name)(da.values, window=7, axis=0,
+                                              min_count=1)
+            self.assertArrayEqual(actual.values, expected)
+
+        # Using center=False
+        rolling_obj = da.rolling(time=7, center=False)
+        for name in ('sum', 'mean', 'std', 'min', 'max', 'median'):
+            actual = getattr(rolling_obj, name)()['time']
+            self.assertDataArrayEqual(actual, da['time'])
+
+        # Using center=True
+        rolling_obj = da.rolling(time=7, center=True)
+        for name in ('sum', 'mean', 'std', 'min', 'max', 'median'):
+            actual = getattr(rolling_obj, name)()['time']
+            self.assertDataArrayEqual(actual, da['time'])
+
+        # catching invalid args
+        with self.assertRaisesRegexp(ValueError, 'Rolling.median does not'):
+            da.rolling(time=7, min_periods=1).median()
+
+    def test_rolling_pandas_compat(self):
+        s = pd.Series(range(10))
+        da = DataArray.from_series(s)
+
+        for center in (False, True):
+            for window in [1, 2, 3, 4]:
+                for min_periods in [None, 0, 1, 2]:
+                    if min_periods is not None and window < min_periods:
+                        min_periods = window
+                    s_rolling = pd.rolling_mean(s, window, center=center,
+                                                min_periods=min_periods)
+                    da_rolling = da.rolling(index=window, center=center,
+                                            min_periods=min_periods).mean()
+                    # pandas does some fancy stuff in the last position
+                    np.testing.assert_allclose(s_rolling.values[:-1],
+                                               da_rolling.values[:-1])
+                    np.testing.assert_allclose(s_rolling.index,
```

Comment:
Can you give a specific example for which there is a discrepancy?

I'm getting a different error when I run your tests:
```

self = <xarray.test.test_dataarray.TestDataArray testMethod=test_rolling_pandas_compat>

    def test_rolling_pandas_compat(self):
        s = pd.Series(range(10))
        da = DataArray.from_series(s)

        for center in (False, True):
            for window in [1, 2, 3, 4]:
                for min_periods in [None, 0, 1, 2]:
                    if min_periods is not None and window < min_periods:
                        min_periods = window
                    s_rolling = pd.rolling_mean(s, window, center=center,
                                                min_periods=min_periods)
                    da_rolling = da.rolling(index=window, center=center,
                                            min_periods=min_periods).mean()
                    # pandas does some fancy stuff in the last position
                    np.testing.assert_allclose(s_rolling.values[:-1],
>                                              da_rolling.values[:-1])

xarray/test/test_dataarray.py:1301:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

x_id = array([ True, False, False, False, False, False, False, False, False], dtype=bool)
y_id = array([False, False, False, False, False, False, False, False, False], dtype=bool), hasval = 'nan'

    def chk_same_position(x_id, y_id, hasval='nan'):
        """Handling nan/inf: check that x and y have the nan/inf at the same
            locations."""
        try:
            assert_array_equal(x_id, y_id)
        except AssertionError:
            msg = build_err_msg([x, y],
                                err_msg + '\nx and y %s location mismatch:'
                                % (hasval), verbose=verbose, header=header,
                                names=('x', 'y'), precision=precision)
>           raise AssertionError(msg)
E           AssertionError:
E           Not equal to tolerance rtol=1e-07, atol=0
E
E           x and y nan location mismatch:
E            x: array([ nan,  0.5,  1.5,  2.5,  3.5,  4.5,  5.5,  6.5,  7.5])
E            y: array([ -9.223372e+18,   5.000000e-01,   1.500000e+00,   2.500000e+00,
E                    3.500000e+00,   4.500000e+00,   5.500000e+00,   6.500000e+00,
E                    7.500000e+00])

/usr/local/anaconda/envs/xarray-dev/lib/python2.7/site-packages/numpy/testing/utils.py:656: AssertionError
```

It looks like there might be a casting issue where int is not getting converted to float properly.